apps/src/studio/StudioAnimation.js

```diff
@@ -176,32 +241,65 @@ StudioAnimation.prototype.redrawCenteredAt = function (center) {
  * Sets which animation to play out of the sprite sheet.
  * Animations are indexed by their position in the sprite sheet, where each
  * animation is its own column and animation zero is the far-left column.
+ * @param {!string} animationType
  * @param {!number} animationIndex
  */
-StudioAnimation.prototype.setCurrentAnimation = function (animationIndex) {
-  this.currentAnimation_ = animationIndex;
+StudioAnimation.prototype.setCurrentAnimation = function (animationType, animationIndex) {
+  this.currentAnimationType_ = animationType;
+  this.currentAnimationIndex_ = animationIndex;
 };
 
 /**
- * Set the animation rate for this item's sprite.
- * @param {number} framesPerSecond
+ * Creates a new special animation types based on specific frames to play from
+ * the sprite sheet.
```

Comment:
description added.

As structured, `StudioSpriteSheet` maintains info about the animations intrinsic to the spritesheet and the associated metadata. `StudioAnimation` maintains info about the runtime-created "special animations" that reference intrinsic animation frames from `StudioSpriteSheet`. I see your point, we could build this directly into `StudioSpriteSheet`, however.